core/base/scalarFieldSmoother/ScalarFieldSmoother.h

```diff
@@ -48,13 +48,26 @@ namespace ttk {
       if(triangulation) {
         triangulation->preconditionVertexNeighbors();
       }
-
       return 0;
     }
+#if TTK_ENABLE_MPI
+    template <class triangulationType>
+    int preconditionDistributedTriangulation(triangulationType *triangulation) {
+      // Pre-condition distributed functions.
+      if(triangulation) {
+        this->printMsg("Preconditioning Vertices");
+        triangulation->preconditionDistributedVertices();
```

Comment:
the calling program should not explicitly call this function.
this should be "hidden".

please move this piece of code to `core/base/triangulation/Triangulation.h`, in the functions `setInputPoints` (called for everything explicit) and `setInputGrid` (called for everything implicit).

Reply:
Here I looked at the non-distributed preconditioning steps and I thought that we don't want to do the preconditioning steps when they are not needed for the filter. Or should we assume that every (distributed) filter which wants a triangulation also will _probably_ need to transfer ghost cell values and therefore need to transform between local and global ids?

Reply:
in short, yes :)